File: XGArqmobWsDemo-Objc/ViewController.m
//
//  ViewController.m
//  XGArqmobWsDemo-Objc
//
//  Created by Vero on 07/05/2020.
//  Copyright © 2020 Vero. All rights reserved.
//
#define kDataKey        @"DATA"
#define kTemporal       @"tmp"
#define kImgKey         @"IMG"
#define kDataFile       @"data.plist"
#define kKmlFile        @"file.kml"
#define kKmlFilePList   @"kml.plist"
#define kMBFile         @"tiles.mb"
#define kMBFileInicio   @"tilesInicio.mb"
#define kMBFileFin      @"tilesFin.mb"
#define kElevation      @"elevation.json"
#import "ViewController.h"
@import XGArqmobWs;
@interface ViewController ()
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    [self downloadData];
}

- (void) downloadData {
    ArqmobService *service = [[ArqmobService alloc] init];
    
    [service getRouteDetailWithLanguage:@"es" nid:@"33778" completionClosure:^(RouteDetail * _Nullable value, NSInteger codeStatus) {
        if (codeStatus == 200) {
            // OK
            NSString *dataPath = [[ViewController OfflineDataDirectoryByID:value.codigo] stringByAppendingPathComponent:kDataFile];
           NSMutableData *data = [[NSMutableData alloc] init];
        
           NSKeyedArchiver *archiver = [[NSKeyedArchiver alloc] initForWritingWithMutableData:data];
           [archiver encodeObject:value forKey:@"roterio"];
           [archiver finishEncoding];
           [data writeToFile:dataPath atomically:YES];
        } else {
            NSLog(@"Error en la petición");
        }
    }];
    
    [service getOriginWithDenominacion:@"monterrei" language:@"es" completionClosure:^(NSArray<Origin *> * _Nullable items, NSInteger codeStatus) {
        NSLog(items.firstObject.imagenes.firstObject.url);
    }];

}
+(NSString *) OfflineDataDirectoryByID: (NSString *) idCode
   {
       NSString *dataDirectory = [self OfflineDataDirectory];
       
       dataDirectory = [dataDirectory stringByAppendingPathComponent :idCode];
       
       NSFileManager *fm = [[NSFileManager alloc] init];
       
       BOOL isDir = false;
       
       if ((![fm fileExistsAtPath:dataDirectory isDirectory:&isDir]) || !isDir)
       {
           [fm createDirectoryAtPath:dataDirectory withIntermediateDirectories:YES attributes:nil error:nil];
       }
       return dataDirectory;
       
   }
+(NSString *) OfflineDataDirectory
{
    NSArray *paths = NSSearchPathForDirectoriesInDomains(NSLibraryDirectory, NSUserDomainMask, YES) ;
    
    NSString *dataDirectory = [[paths objectAtIndex:0]copy];// [[NSBundle mainBundle] resourcePath];
    
    dataDirectory = [dataDirectory stringByAppendingPathComponent :kDataKey];
    
    NSFileManager *fm = [[NSFileManager alloc] init];
     
    BOOL isDir = false;
    
    if ((![fm fileExistsAtPath:dataDirectory isDirectory:&isDir]) || !isDir)
    {
        [fm createDirectoryAtPath:dataDirectory withIntermediateDirectories:YES attributes:nil error:nil];
    }
    return dataDirectory;
    
}
@end

File: XGArqmobWsDemo/ViewController.swift
//
//  ViewController.swift
//  XGArqmobWsDemo
//
//  Created by Vero on 07/05/2020.
//  Copyright © 2020 Vero. All rights reserved.
//

import UIKit
import XGArqmobWs

class ViewController: UIViewController {
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
//        getRecursosByType(-8.545729692550367, latitude: 42.88085063626383, radio: "10000", numResults: "50", filterType: [16,15,24,18,20,1,5,22,27,17,6,13,9,7,28,14,23,8,2,11,21]) { (items, status) in
//            if let places = items, places.count > 0 {
//            }
//        }
        
        
        getOriginWith("monterrei") { (items, status) in
            items
        }
    }
    
    func getRecursosByType(_ longitude: Double, latitude: Double, radio: String, numResults: String, filterType: Array<Int64>? , completion: @escaping (_ responses: Array<Place>?, _ codeStatus:ConstantsStatus)-> ())  {
        let userLocation = Location()
        userLocation.latitud = "\(latitude)"
        userLocation.longitud = "\(longitude)"
        let lang = "es-gl"
        ArqmobService.shared.getPlaces(location: userLocation, language: lang, radio: radio, numResults: numResults, filterTypes: filterType) { (items, status) in
            completion(items, status)
        }
    }
    
    func getOriginWith(_ denominacion: String, completion: @escaping (_ responses: Array<Origin>?, _ codeStatus:ConstantsStatus)-> ())  {
        ArqmobService.shared.getOriginWith() { (items, status) in
            completion(items, status)
        }
    }
}

